tools/junit-openfeature/src/main/java/dev/openfeature/contrib/tools/junitopenfeature/TestProvider.java

```diff
@@ -0,0 +1,128 @@
+package dev.openfeature.contrib.tools.junitopenfeature;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Reason;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.FlagNotFoundError;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.OpenFeatureError;
+import dev.openfeature.sdk.exceptions.ProviderNotReadyError;
+import dev.openfeature.sdk.exceptions.TypeMismatchError;
+import dev.openfeature.sdk.providers.memory.Flag;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * TestProvider based on InMemoryProvider but with another dimension added to the maps of flags.
+ */
+@Slf4j
+public class TestProvider extends EventProvider {
```

Comment:
nit - any chance to modify SDK InMemoryProvider to get all these features ? IMO that would be beneficial with maintenance.

Reply:
I am thinking of changing this provider to a wrapper like the multi-provider, which utilizes multiple InMemory instances. Do you think this would be a good idea?

Reply:
do you think the inmemory approach is better?

Reply:
The inmem provider was created to help SDK testing. And there's no harm in exposing some extra methods to help with testing. The only ask is to keep inmem provider with SDK so we don't get a circular dependency between SDK & contribs.